aiohttp_json_rpc/protocol.py

```diff
@@ -176,3 +183,27 @@ def encode_error(error, id=None):
         msg['error']['data'] = error.data
 
     return json.dumps(msg)
+
+
+def decode_error(msg: JsonRpcMsg):
+    error_code = msg.data['error']['code']
+
+    # We are guaranteed to hit one of the following if statements
+    # Please see PR #14 for more information.
+    exception = None
+
+    if error_code == RpcInvalidRequestError.ERROR_CODE:
```

Comment:
it's better to do a lookup table for this

Reply:
How would you format that lookup table? The only reason I didn't is because I'm imagining something like the below and it just seemed a bit too awkward

```py
lookup_table = {
    RpcInvalidRequestError.ERROR_CODE: RpcInvalidRequestError,
    RpcMethodNotFoundError.ERROR_CODE: RpcMethodNotFoundError,
    ...
}
```

Reply:
```python
error_map = {
    e.ERROR_CODE: e
    for e in (RpcInvalidRequestError, RpcMethodNotFoundError, ...)
}
```

Maybe if there were a base exception class you could do it in more generic way via iterating subclasses.

Reply:
Okay, now I see there's a base class (honestly, it would be better if it inherited some `IOError`), so the snippet above turns into:
```python
from .exceptions import RpcError

error_map = {
    e.ERROR_CODE: e
    for e in RpcError.__subclasses__()
}
```

You could go further and turn this into a lazy class property of `RpcError`

Reply:
Didn't realize dunder subclasses was a thing, that's actually a really cool solution.

Reply:
It appeared in python 3